spyder/widgets/status.py

```diff
@@ -52,20 +60,6 @@ def __init__(self, parent, statusbar, icon=None):
         self.label_value.setAlignment(Qt.AlignRight)
         self.label_value.setFont(self.text_font)
 
-        # Layout
-        layout = QHBoxLayout()
-        layout.setSpacing(0)  # Reduce space between icon and label
-        layout.addWidget(self.label_icon)
-        self.label_icon.setVisible(icon is not None)
```

Comment:
Why are you removing this condition?

Reply:
Because this condition is already applied in `set_icon`